### DISCUSSION TITLE
How to add type casting to a custom field

### DISCUSSION THREAD

**Discussion by JilsonSaji** [filamentphp/filament#11760] [Help] [answered]
### Package

Panel builder

### Package Version

v3.0.0

### How can we help you?

I have a custom field that take an array as an input and it should return a json string 
eg:

`CustomField::make('my_field')->items(array);`

I am expecting a string on 'my_field' property in the DB

Anyone have any suggestions?

**Comment by leandrocfe** [comment, marked as answer]
Add in your Model `array` or `json`
```php
protected $casts = [
    'my_field' => 'array'
];
```

**Reply by JilsonSaji** [reply]
Thank you. It worked.
I wanted to mutate the data from the CustomField...!
Regards
